addons/ccm-aws/ccm-aws.yaml

```diff
@@ -165,7 +165,26 @@ spec:
             - --cloud-provider=aws
             - --use-service-account-credentials=false
             - --configure-cloud-routes=false
+            - --cloud-config=/etc/kubernetes/cloud/config
             - --v=2
           resources:
             requests:
               cpu: 200m
+          volumeMounts:
+            - mountPath: /etc/kubernetes/cloud
+              name: cloud-config
+              readOnly: true
+      volumes:
+        - name: cloud-config
+          secret:
+            defaultMode: 420
+            secretName: cloud-config
+---
+apiVersion: v1
+data:
+  config: {{ awsCCMCloudConfig .Config.Name .Config.ClusterNetwork.IPFamily | b64enc }}
+kind: Secret
+metadata:
+  name: cloud-config
```

Comment:
We tend to allow users to provide their own cloud-config whenever that's possible. I think we should do the same here. That being said, the cloud-config should be taken from the API (`.cloudProvider.cloudConfig`). We can default cloud-config there if needed.